src/main/scala/intellij/haskell/repl/HaskellConsoleExecuteActionHandler.scala

```diff
@@ -0,0 +1,63 @@
+package intellij.haskell.repl
+
+import java.io.IOException
+
+import com.intellij.execution.console.LanguageConsoleImpl
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.openapi.application.Result
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.project.{Project, ProjectManager}
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.text.StringUtil
+import intellij.haskell.HaskellNotificationGroup
+
+class HaskellConsoleExecuteActionHandler(val project: Project, var processHandler: ProcessHandler) {
+  def runExecuteAction(console: LanguageConsoleImpl, executeImmediately: Boolean) {
+    if (executeImmediately) {
+      execute(console)
+    } else {
+      val editor = console.getCurrentEditor
```

Comment:
I think ActionUtil can not be used in this place since there does not exist an `AnActionEvent` instance.